### PR TITLE
Move to github actions, test current napari master during CI

### PR DESCRIPTION
**Pull request by tlambert03**
This adds one more test to the CI here, that pulls the current master branch of napari and runs it against the current napari-plugin-engine version.  (to give us a heads up if we've changed something that will break napari tests).  currently,... I think there's one broken thing in the plugin error reporter in napari master.

### CODE REVIEW

**Comment by tlambert03** [comment]
the fix could be on either side, the error is just that fetching metadata for a package that does not exist is now an error.  And on of the napari tests fetches a non-existent plugin:

```pytb
________________________________________________________________________________
Traceback (most recent call last):
  File "/private/tmp/.tox/napari/lib/python3.8/site-packages/napari/_qt/qt_plugin_report.py", line 157, in set_plugin
    err_string = format_exceptions(plugin, as_html=True)
  File "/private/tmp/.tox/napari/lib/python3.8/site-packages/napari/plugins/exceptions.py", line 49, in format_exceptions
    package_meta = standard_metadata(err0.plugin)
  File "/private/tmp/.tox/napari/lib/python3.8/site-packages/napari_plugin_engine/dist.py", line 142, in standard_metadata
    raise ValueError(f"could not find metadata for {plugin}")
ValueError: could not find metadata for mock
```

need to get xvfb working on tests first

**Comment by tlambert03** [comment]
I used this PR to make setup.cfg match the patterns used in napari, and I moved from travis to github actions for testing and releases.
I reverted the one change that caused one napari test in master to fail (when requesting metadata for a non-existent package).  So with this PR, the install should be working fine, and we're testing against the napari master branch with all future PRs.

**Comment by tlambert03** [comment]
@jni, if you can take a quick look and merge if you're happy, I'll check to see whether this fixes #8, and then push a new release to PyPI with the dask-delayed fixes

File: napari_plugin_engine/dist.py [hunk 1]
```diff
@@ -139,7 +139,9 @@ def standard_metadata(plugin: Any) -> Dict[str, Optional[str]]:
     if not get_dist(plugin):
         _top_level_module_to_dist.cache_clear()
         if not get_dist(plugin):
-            raise ValueError(f"could not find metadata for {plugin}")
+            return {}
+            # TODO: decide appropriate behavior here.
+            # raise ValueError(f"could not find metadata for {plugin}")
```
Comment (jni) [inline napari_plugin_engine/dist.py:144]: What's the story here?

File: pyproject.toml [hunk 2]
```diff
@@ -1,13 +1,15 @@
-# pyproject.toml
 [build-system]
 requires = [
-    "setuptools>=42",
-    "wheel",
-    "setuptools_scm[toml]>=3.4"
+  "setuptools >= 42",
+  "wheel",
+  "setuptools_scm[toml]>=3.4"
```
Comment (jni) [inline pyproject.toml:5]: Is there supposed to be a space around `>=` or not?

File: tox.ini [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
 [testenv:linting]
 skip_install = true
-basepython = python3.6
 deps = pre-commit
 commands = pre-commit run --all-files --show-diff-on-failure
 
+[testenv:napari]
+basepython = python3.8
+deps = 
+    -rhttps://raw.githubusercontent.com/napari/napari/master/requirements/test.txt
```
Comment (jni) [inline tox.ini:33]: Can we put a space after the `r`? I thought this was a typo for a second

File: napari_plugin_engine/dist.py (hunk 1 above)
Comment (jni) [reply napari_plugin_engine/dist.py:144]: nvmd, just read above! 😅

Comment (tlambert03) [reply napari_plugin_engine/dist.py:144]: yeah, that's the thing I "broke" ... made it an exception to ask for metadata from an uninstalled package.  just decided to kick that can down the road and let it pass for now.

File: tox.ini (hunk 3 above)
Comment (tlambert03) [reply tox.ini:33]: this seems to be how they[ document it](https://tox.readthedocs.io/en/latest/example/basic.html#depending-on-requirements-txt-or-defining-constraints)... but I just tried with a space and it also works.  weird.

File: pyproject.toml (hunk 2 above)
Comment (tlambert03) [reply pyproject.toml:5]: it appears to not matter... but seems like I should at least be consistent! 😂 
i removed the space

**Comment by tlambert03** [comment]
until this merges, github actions tests are here: https://github.com/tlambert-forks/napari-plugin-engine/actions/runs/185018779

will merge after they pass this last time

**Pull request merged**